Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -476,14 +477,59 @@ class InAppMessageViewTest: IntegrationTest {
         XCTAssertEqual(inlineActionDelegate1.onActionClickReceivedArguments?.actionValue, "Test")
         XCTAssertEqual(inlineActionDelegate1.onActionClickReceivedArguments?.actionName, "")
     }
+
+    // MARK: - Deeplinks
+
+    @MainActor
+    func test_deeplinks_givenButtonTappedWithValidDeeplink_expectOpenDeeplink() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+        onCustomActionButtonPressed(onInlineView: inlineView, isSystem: true, action: "https://customer.io")
+
+        // Since a system call, hence no delegate is called
+        XCTAssertFalse(inlineMessageDelegateMock.onActionClickCalled)
+
+        // Do not dismiss inline message when deep link is opened
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // If url is valid, check if `handleDeepLink` method is called
+        XCTAssertTrue(deeplinkUtilMock.handleDeepLinkCalled)
+        XCTAssertEqual(deeplinkUtilMock.handleDeepLinkReceivedArguments?.absoluteString, "https://customer.io")
+    }
+
+    @MainActor
+    func test_deeplinks_givenButtonTappedWithInValidDeeplink_expectOpenDeeplink() async {
+        let givenInlineMessage = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage]
+
+        let inlineView = InAppMessageView(elementId: givenInlineMessage.elementId!)
+        await onDoneRenderingInAppMessage(givenInlineMessage, insideOfInlineView: inlineView)
+
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+        onCustomActionButtonPressed(onInlineView: inlineView, isSystem: true, action: "ht!tp://invalid-url")
+
+        // Since a system call, hence no delegate is called
+        XCTAssertFalse(inlineMessageDelegateMock.onActionClickCalled)
+
+        // Do not dismiss inline message when deep link is opened
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // If url is valid, check if `handleDeepLink` method is called
+        XCTAssertFalse(deeplinkUtilMock.handleDeepLinkCalled)
+    }
 }
 
 @MainActor
 extension InAppMessageViewTest {
-    func onCustomActionButtonPressed(onInlineView inlineView: InAppMessageView) {
+    func onCustomActionButtonPressed(onInlineView inlineView: InAppMessageView, isSystem: Bool = false, action: String = "Test") {
         // Triggering the custom action button on inline message from the web engine
         // mocks the user tap on custom action button
-        getWebEngineForInlineView(inlineView)?.delegate?.tap(name: "", action: "Test", system: false)
+        getWebEngineForInlineView(inlineView)?.delegate?.tap(name: "", action: action, system: isSystem)
     }
```

Comment:
I find this function when it's being used for clicking custom action buttons and deep links hard to read. 

I think it would be easier if there were 2 separate functions: 
1. `onCustomActionButtonPressed(onInlineView inlineView: InAppMessageView, action: String)`
2. `onDeepLinkActionButtonPressed(onInlineView inlineView: InAppMessageView, deepLink: URL)`

Reply:
I do not mind having two methods but the working is same, just the parameter values are changing.